lib/Target/RISCV/MCTargetDesc/RISCVAsmBackend.cpp

```diff
@@ -85,15 +87,26 @@ class RISCVAsmBackend : public MCAsmBackend {
 };
 
 bool RISCVAsmBackend::writeNopData(uint64_t Count, MCObjectWriter *OW) const {
-  // Once support for the compressed instruction set is added, we will be able
-  // to conditionally support 16-bit NOPs
-  if ((Count % 4) != 0)
-    return false;
-
+  bool HasC = STI->getFeatureBits()[RISCV::FeatureC];
```

Comment:
Can we add an MC-layer test that will exercise this?

Reply:
Add cnop.s MC testcase to test c.nop generation.